tests/Game/Board.tests.cpp

```diff
@@ -1,69 +1,81 @@
 #include <gtest/gtest.h>
 #include <Minefield/Game/Board.h>
+#include <Minefield/Console/Output.h>
 
 namespace game::tests
 {
     TEST(Board, constructor_should_keep_dimensions_and_tiles)
     {
         BoardDimensions dimensions(3, 3);
-        utils::board::TileMatrix matrix(3, std::vector<Tile>(3));
+        utils::board::TileMatrix matrix(3, std::vector<TileType>(3));
 
         for (unsigned int x = 0; x < dimensions.x; ++x)
         {
             for (unsigned int y = 0; y < dimensions.y; ++y)
             {
-                matrix[x][y] = Tile(TileType::Empty, {x, y});
+                matrix[x][y] = TileType::Empty;
             }
         }
 
-        matrix[1][1] = Tile(TileType::Bomb, {1, 1});
+        matrix[1][1] = TileType::Bomb;
 
         Board board(matrix, dimensions);
 
         EXPECT_EQ(board.dimensions().x, 3);
         EXPECT_EQ(board.dimensions().y, 3);
 
-        utils::board::TileMatrix boardMatrix = board.matrix();
-        EXPECT_EQ(boardMatrix[0][0].tileType, TileType::Empty);
-        EXPECT_EQ(boardMatrix[1][1].tileType, TileType::Bomb);
-        EXPECT_EQ(boardMatrix[2][2].tileType, TileType::Empty);
+        const TileType* tile1 = game::utils::board::getTile(Coordinate(0, 0), matrix, dimensions);
+        const TileType* tile2 = game::utils::board::getTile(Coordinate(1, 1), matrix, dimensions);
+        const TileType* tile3 = game::utils::board::getTile(Coordinate(2, 2), matrix, dimensions);
+
+        EXPECT_EQ(*tile1, TileType::Empty);
+        EXPECT_EQ(*tile2, TileType::Bomb);
+        EXPECT_EQ(*tile3, TileType::Empty);
     }
 
     TEST(Board, change_tile_type_should_only_update_target_tile)
     {
         BoardDimensions dimensions(5, 5);
-        utils::board::TileMatrix matrix(5, std::vector<Tile>(5, Tile(TileType::Empty, {})));
-        Board board(matrix, dimensions);
+        utils::board::TileMatrix matrix;
+        Board board(utils::board::TileMatrix(), BoardDimensions(5, 5));
 
-        Coordinate coordToChange = {2, 3};
-        board.changeTileType(coordToChange, TileType::Guess);
+        for (unsigned int x = 0; x < dimensions.x; ++x)
```

Comment:
#must probably invert the iteration, first `y` then `x`, and flip the access in line 46